### DISCUSSION TITLE
Database Searching/PDF searching

### DISCUSSION THREAD

**Discussion by bji219** [danny-avila/LibreChat#543] [Feature Requests & Suggestions]
Using LangChain would it be possible to "teach" one of the LLMs a large pdf document such as a book or technical manual? The end result would be a searchable database, and the ability to access the knowledge by using natural language prompts. Would something like this be possible through a plug-in?

**Comment by danny-avila** [comment]
Already started work on this, should be the next biggest feature

**Reply by bji219** [reply]
Awesome! Is there any way to contribute ?

**Reply by fuegovic** [reply]
For a more direct discussion on the subject you can  join us on [discord](https://discord.gg/wBbdrTtw5F) , look for this post "Langchain / Document Upload/ VectorDB Ideas" in #suggestions

**Comment by MatthiasBergner** [comment]
Are there any news for this topic? I was wondering if we could just recycle some code of the RAG project. Since RAG is only used for a specific user and chat, it is limited in this way. As this question #543 suggests, we would just need another behavior. We need to upload the documents from the admin side and give the RAG the possibility to access the database and documents with all users and any chat. So It would be an "open" database for general knowledge the company provides for every user and chat - and a "closed" for the standard behavior we have right now - for the specific user and chat.
